### ISSUE TITLE
Move `GraphQLApi` component to `fuel-indexer-api-server`

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#175]
- The `GraphQLApi` component is tucked in the `fuel-indexer` crate
- The `GraphQLApi` component depends `fuel-indexer-database-types`, `fuel-indexer-lib`, and `fuel-indexer-schema`
- We should move the `GraphQLApi` component to `fuel-indexer-api-server`, which already depends on `fuel-indexer-schema`, `fuel-indexer-lib`, and `fuel-indexer`
  - At most, `fuel-indexer-api-server` would have 1 more dependency (i.e., `fuel-indexer-database-types`)
  - This would in turn cause `fuel-indexer` to depend on `fuel-indexer-api-server` (reverses the dependency)
  - Would make `fuel-indexer-api-server` more of an independent crate

Additional context
- This might require a bit more thought

**Comment by ra0x3** [comment]
<img width="749" alt="Screen Shot 2022-10-11 at 12 00 32 PM" src="https://user-images.githubusercontent.com/17253182/195142131-320463a4-1b50-4ac1-86c5-ebd9bfdfc818.png">

**Comment by deekerno** [comment]
The intention of this effort is to make `fuel-indexer-api-server` into a more independent crate. Upon moving `api.rs` to `fuel-indexer-api-server`, one has to change the dependencies and imports of `fuel-indexer-api-server` in order to reverse the dependency upon `fuel-indexer` and instead make the latter depend on the former. As such, I've moved the `Config` and `Manifest` structs to `fuel-indexer-lib`; they're already used in a few other places as well so it seems like a sensible choice.

However, `api.rs` imports `IndexerError` for use in the `ApiError` enum; `IndexerError` currently resides in the `fuel-indexer` crate. Additionally, the `IndexerError` enum uses types from other parts of the `fuel-indexer` crate, e.g. `ffi`, `executor`, etc. These pieces will need to be moved as well in order for `fuel-indexer-api-server` to be independent of `fuel-indexer`. One should also note that the API uses `SchemaManager` in some of the endpoint handlers; this also lives in `fuel-indexer` and uses the local `ffi` crate as well.

tl;dr - where should we relocate `IndexerError`, `SchemaManager`, and their necessary local imports such that they can be imported into `fuel-indexer-api-server`?

**Comment by ra0x3** [comment]
@deekerno Hmm I'm thinking 🤔 
- Maybe move `SchemaManager` into `fuel-indexer-schema` 
  - It seems super small so shouldn't be any issues
- `IndexerError` is a little more tricky since it seems to be legit needed by `fuel-indexer` and `fuel-indexer-api-server`
- I do think `IndexerError` is a sort've turning into a [God object](https://en.wikipedia.org/wiki/God_object) 👎 
- I feel like if `IndexerError` in `api.rs` is the only blocker left, maybe just remove it from `ApiError` ?
  - Web routes will still be infallible and will return generic error message [we just won't be able to handle them](https://github.com/FuelLabs/fuel-indexer/blob/master/fuel-indexer/src/api.rs#L92)
  - `ApiError` also still can/will derive `SqlxError`,  `IoError`, `DatabaseError`, and `Unknown` so it will still have a lot of overlap with `IndexerError`

Lmk what you think

**Comment by deekerno** [comment]
Totally agree with moving `SchemaManager` into `fuel-indexer-schema` and with removing `IndexerError` from `ApiError`. 👍🏽 I'll update the PR.

**Comment by ra0x3** [comment]
Reopening since we had to revert this
